File: client/src/types/character.ts
export interface CharacterStats {
  health: number;
  intelligence: number;
  charisma: number;
  wealth: number;
  happiness: number;
  luck: number;
}

export interface Character {
  id: number;
  name: string;
  gender: 'male' | 'female';
  age: number;
  stage: string;
  is_alive: boolean;
  city: { id: number; name: string } | null;
  stats: CharacterStats | null;
}

export interface CreateCharacterRequest {
  name: string;
  gender: 'male' | 'female';
  city_id: number;
}

export interface PendingEvent {
  id: number;
  title: string;
  description: string;
  choices: { index: number; text: string }[];
}

export interface AdvanceYearResponse {
  character: { age: number; stage: string; is_alive: boolean };
  year_summary: { stat_changes: Record<string, number>; stage_changed: boolean; death_cause: string | null };
  pending_events: PendingEvent[];
}

File: server/app/schemas/character.py
from pydantic import BaseModel, Field
from typing import Optional


class CreateCharacterRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=50, description="角色姓名")
    gender: str = Field(..., pattern="^(male|female)$", description="性别")
    city_id: int = Field(..., gt=0, description="出生城市ID")


class CharacterStatsResponse(BaseModel):
    health: int
    intelligence: int
    charisma: int
    wealth: int
    happiness: int
    luck: int


class CityBrief(BaseModel):
    id: int
    name: str


class CharacterResponse(BaseModel):
    id: int
    name: str
    gender: str
    age: int
    stage: str
    is_alive: bool
    city: Optional[CityBrief] = None
    stats: Optional[CharacterStatsResponse] = None


class LifeRecordResponse(BaseModel):
    id: int
    age: int
    year: int
    summary: str
    stat_changes: Optional[dict] = None


class AdvanceYearResponse(BaseModel):
    character: dict
    year_summary: dict
    pending_events: list
